rts/lra/coordinator/src/main/java/io/narayana/lra/coordinator/domain/model/LongRunningAction.java

```diff
@@ -919,7 +916,15 @@ private int scheduleCancelation(Runnable runnable, Long timeLimit) {
             finishTime = LocalDateTime.now(ZoneOffset.UTC).plusNanos(timeLimit * 1000000);
         }
 
-        scheduledAbort = scheduler.schedule(runnable, timeLimit, TimeUnit.MILLISECONDS);
+        try {
+            scheduledAbort = scheduler.schedule(runnable, timeLimit, TimeUnit.MILLISECONDS);
+        } catch (RejectedExecutionException executionException) {
+            updateState(LRAStatus.Cancelling);
+            LRALogger.logger.warnf(
```

Comment:
Logging the exception message is vital for diagnosis of the underlying cause. Note that I recently fixed JBTM-3525 (include exception messages in log output) and would not want to reopen it ;-)

Note that the failure is not fatal since, when periodic recovery runs it will eventually discover that it is eligible for cancellation. It might be worthwhile adding that comment in the catch block for future maintainers of this code.

Reply:
+1 I will update the PR. Thanks